### DISCUSSION TITLE
Request for examples/docs

### DISCUSSION THREAD

**Discussion by weeping-somnambulist** [questnet/free-socks#6] [General]
Hello! I found this crate and tried it out, and I was able to get some basic functionality working with the Freeswitch Inbound ESL. I've been struggling to send an SMS message and subscribe to events, among other things. 

Is there any chance you can give me a few examples? I can write some docs for the readme if you'd like. 

Let me know if you'd like to chat about this via email. I'd like to help with this crate, I'm just struggling to understand exactly how to implement it, and where it's feature boundary stops. 

Any help would be appreciated!

**Comment by pragmatrix** [comment]
Hi, this is an early draft of what could be some day a complete Rust interface to FreeSWITCH event sockets. We do run a small watchdog service based on the code base, but realized recently that we need a robust concept for detecting of socket disconnects. This can be solved by keepalives and an exposed future that completes when the socket disconnects which I am trying to implement in #4. 

Obviously, this is work in progress, and not something I can recommend to run in production yet. Furthermore,  features are incomplete and the API surface will most likely change and so documentation isn't a priority. Only simple commands are supported.

One example is in the examples subdirectory (just a client that does authentication), but you can use simple api commands through `EventSocket::command` and `EventSocket::api`, see also the implementations of `hupall`, `channels`, and `channels_count` async functions. `bgapi` and `application` requests are not supported.

The future of this project depends on our commitment to move away from NEventSocket to Rust and I am not sure if and when this happens.
